climada/hazard/landslide.py

```diff
@@ -188,304 +46,174 @@ def __init__(self):
-            self._intensity_prob_to_binom(n_years)
-            self.check()
-
-            if incl_neighbour:
-                LOGGER.info('Finding neighbouring pixels...')
-                self.centroids.set_meta_to_lat_lon()
-                self.centroids.set_geometry_points()
-                self._intensity_binom_to_range(max_dist)
-                self.check()
-
-            if check_plots == 1:
-                fig1 = plt.subplots(nrows=1, ncols=1)[0]
-                self.plot_raw()
-                fig1.suptitle('Raw data: Occurrence prob of LS per year', fontsize=14)
-
-                fig2 = plt.subplots(nrows=1, ncols=1)[0]
-                self.plot_events()
-                fig2.suptitle('Prob. LS Hazard Set n_years = %i' % n_years, fontsize=14)
-
-            return self
-
-        elif ls_model == "NASA":
-            if not bbox:
-                LOGGER.error('Empty bounding box, please set bounds.')
-                raise ValueError()
-
-            if not path_sourcefile:
-                LOGGER.error('Empty sourcefile, please specify')
-                raise ValueError()
-            window_array = self._get_window_from_coords(path_sourcefile, bbox)
-            pixel_height, pixel_width = self._get_raster_meta(path_sourcefile, window_array)
-            self.set_raster([path_sourcefile], window=Window(window_array[0], window_array[1],
-                                                             window_array[3], window_array[2]))
-            LOGGER.info('Setting probability values from categorical landslide hazard levels...')
-            self._intensity_cat_to_prob(max_prob)
-            self.centroids.set_raster_from_pix_bounds(bbox[0], bbox[3], pixel_height, pixel_width,
-                                                      window_array[3], window_array[2])
-            LOGGER.info('Generating binary landslides...')
-            self._intensity_prob_to_binom(n_years)
-            self.check()
-
-            if incl_neighbour:
-                LOGGER.info('Finding neighbouring pixels...')
-                self.centroids.set_meta_to_lat_lon()
-                self.centroids.set_geometry_points()
-                self._intensity_binom_to_range(max_dist)
-                self.check()
-
-            if check_plots == 1:
-                fig1, _ = plt.subplots(nrows=1, ncols=1)
-                self.plot_raw()
-                fig1.suptitle('Raw data: Occurrence prob of LS per year', fontsize=14)
-
-                fig2, _ = plt.subplots(nrows=1, ncols=1)
-                self.plot_events()
-                fig2.suptitle('Prob. LS Hazard Set n_years = %i' % n_years, fontsize=14)
-
-            return self
-
-        else:
-            LOGGER.error('Specify the LS model to be used for the hazard-set '
-                         'generation as ls_model=str')
-            raise KeyError
+        for i, j in zip(*self.fraction.nonzero()):
+            self.fraction[i, j] = binom.rvs(n=n_years,
```

Comment:
The mathematics of fraction and/or frequency calculation should be checked. I think it might be `binom.rvs(n=n_years,p=self.fraction[i, j])/n_years`

Reply:
As is in the current code, they are still integer values (number of "successes" or landslide occurrences per grid cell over the whole time period). As the whole grid cell is either unaffected (0) or one to several times, this seemed to be rasonable in the fraction.

In `self.frequency`, I then divide the fraction value by `n_years`. 

So this should be consistent for the impact calculation, right?